Tests/ConfigurationTests/ConfigurationManagerTests.swift

```diff
@@ -0,0 +1,201 @@
+    }
+
+    func removeObject(forKey defaultName: String) {
+        defaults.removeValue(forKey: defaultName)
+    }
+
+    func clearAll() {
+        defaults.removeAll()
+    }
+}
+
+final class ConfigurationManagerTests: XCTestCase {
+
+    // Shared "UserDefaults" to mock app group defaults
+    var sharedDefaults = MockDefaults()
+
+    override func setUp() {
+        APIRequest.Headers.setUserAgent("")
+        Configuration.setURLProvider(MockConfigurationURLProvider())
+        sharedDefaults.clearAll()
+        MockStoreWithStorage.clearTempConfigs()
+    }
+
+    func makeConfigurationFetcher(store: ConfigurationStoring,
+                                  validator: ConfigurationValidating = MockValidator()) -> ConfigurationFetcher {
+        let testConfiguration = URLSessionConfiguration.default
+        testConfiguration.protocolClasses = [MockURLProtocol.self]
+        return ConfigurationFetcher(store: store,
+                                    validator: validator,
+                                    urlSession: URLSession(configuration: testConfiguration))
+    }
+
+    func makeConfigurationManager(name: String? = nil) -> MockConfigurationManager {
+        let configStore = MockStoreWithStorage(etagStorage: sharedDefaults)
+        let manager = MockConfigurationManager(fetcher: makeConfigurationFetcher(store: configStore),
+                                               store: configStore,
+                                               defaults: sharedDefaults)
+        manager.name = name
+        return manager
+    }
+
+    func testWhenConfigIsFetchedAndStoredDependencyIsUpdated() async {
+        let configurationManager = makeConfigurationManager()
+
+        let configData = Data("Privacy Config".utf8)
+        MockURLProtocol.requestHandler = { _ in (HTTPURLResponse.ok, configData) }
+        await configurationManager.refreshNow()
+
+        XCTAssertEqual(configurationManager.dependencyProvider.privacyConfigData, configData)
+        XCTAssertEqual(configurationManager.dependencyProvider.privacyConfigEtag, HTTPURLResponse.testEtag)
+    }
+
+    func testWhenConfigIsNotModifiedThenDependencyIsNotUpdated() async {
+        let configurationManager = makeConfigurationManager()
+
+        MockURLProtocol.requestHandler = { _ in (HTTPURLResponse.notModified, nil) }
+        await configurationManager.refreshNow()
+
+        XCTAssertNil(configurationManager.dependencyProvider.privacyConfigData)
+        XCTAssertNil(configurationManager.dependencyProvider.privacyConfigEtag)
+    }
+
+    func testWhenManagerAIsUpdatedManagerBIsAlsoUpdated() async throws {
+        let managerA = makeConfigurationManager(name: "A")
+        let managerB = makeConfigurationManager(name: "B")
+
+        let configData = Data("Privacy Config".utf8)
+        MockURLProtocol.requestHandler = { _ in (HTTPURLResponse.ok, configData) }
+        await managerA.refreshNow()
+        try await Task.sleep(interval: 2)
```

Comment:
There may be a way we can instead observe the changes we're waiting for, and shortcut the need to sleep - I'll think on it a bit.